source/client/process_impl.cc

```diff
@@ -238,10 +240,12 @@ ProcessImpl::mergeWorkerStatistics(const StatisticFactory& statistic_factory,
 }
 
 void ProcessImpl::createBootstrapConfiguration(envoy::config::bootstrap::v2::Bootstrap& bootstrap,
-                                               const Uri& uri, int number_of_clusters) const {
+                                               const std::vector<UriPtr>& uris,
+                                               int number_of_clusters) const {
   for (int i = 0; i < number_of_clusters; i++) {
     auto* cluster = bootstrap.mutable_static_resources()->add_clusters();
-    if (uri.scheme() == "https") {
+    RELEASE_ASSERT(!uris.empty(), "illegal configuration with zero endpoints");
+    if (uris[0]->scheme() == "https") {
```

Comment:
A comment explaining why all URIs are guaranteed to have the same scheme might be good here.